### ISSUE TITLE
Dealing with circular references and JSON.stringify

### ISSUE THREAD

**Issue by jdus** [antoniandre/splitpanes#38]
Hi,
I have also run into problems with JSON.stringify not allowing circular references. This leads to an error because it is used when computing the slots to export in the render function.
I'm using quasar and to me it happened when I was using a custom quasar-directive "v-scroll" which is translated internally to "__qscroll". Obviously this is not part of the "discardProps" array, which holds all the properties to ignore when calling JSON.stringify.

I reckon this will happen a lot to people when using other frameworks and obviously you cannot keep adding these properties to "discardProps". I have no solution yet, but wanted to put this issue to discussion.
What if cycle.js is used to allow circular references?

Greetings,
John

**Comment by antoniandre** [comment]
Hi @jdus,
Thanks for your feedback.
You are right, you understood the whole problem at stake.
When I started this library I thought it would be great if users could write whatever they want in a splitpanes wrapper and don't bother with splitters.
To allow that, I add the splitters on the fly in a render function. As this is modifying the DOM, there was no problem until someone discovered the reactivity was not deep.

By adding reactivity it ended up in infinite loop of course because every time a slot content changes it triggers the render function which adds splitters, so slots have changed and so on...
That is why I added a JSON flattened structure to compare original and new version on slots change.
To avoid circular reference in JSON.stringify, we have to remove all the instances of vue components, vnodes and this kind of stuffs.

Now I see this path as a dead-end and that's the current limitation of this library.
The only solution I have in mind, but I don't have time for that at the moment, is to remove the automatic splitters creation and let users add them between their panes inside the splitpanes tag.
We would then be able to get rid of the JSON.stringify comparison that causes troubles.
